include/ignition/math/SphericalCoordinates.hh

```diff
@@ -91,15 +95,31 @@ namespace ignition
       public: ~SphericalCoordinates();
 
       /// \brief Convert a Cartesian position vector to geodetic coordinates.
-      /// \param[in] _xyz Cartesian position vector in the world frame.
+      /// This performs a `PositionTransform` from LOCAL to SPHERICAL.
+      ///
+      /// There's a known bug with this computation that can't be fixed on
+      /// version 6 to avoid behaviour changes. Directly call
```

Comment:
@scpeters The comments indicate that the behavior was only intended to be kept for gz-math6. However, it slipped through to gz-math7. Is there some will to straighten this up in 8, or will we live with this forever?

Reply:
sure, we can fix it in gz-math8

can you open an issue or pull request to `main`?

Reply:
Good, I'll try the PR.

I've noticed another discrepancy in the interface: SphericalFromLocalPosition returns spherical coordinates in degrees, while PositionTransform uses radians. Shouldn't this be also united to using degrees everywhere? Or was there a reason for keeping it this way?

Reply:
I think consistency would be preferred; it was probably oversight in the first place that caused them to diverge. A `std::vector<math::Angle>` type would handle the units more clearly but it a little more overhead than `Vector3d`

Reply:
Okay, I've sent the PRs: #596, #597.